adafruit_wm8960.py

```diff
@@ -21,48 +25,1164 @@
+    pass
+
+# I2C address
+_DEFAULT_I2C_ADDR = const(0x1A)
+
+# WM8960 register addresses
+_REG_LEFT_INPUT_VOLUME = const(0x00)
+_REG_RIGHT_INPUT_VOLUME = const(0x01)
+_REG_LOUT1_VOLUME = const(0x02)
+_REG_ROUT1_VOLUME = const(0x03)
+_REG_CLOCKING_1 = const(0x04)
+_REG_ADC_DAC_CTRL_1 = const(0x05)
+_REG_ADC_DAC_CTRL_2 = const(0x06)
+_REG_AUDIO_INTERFACE_1 = const(0x07)
+_REG_CLOCKING_2 = const(0x08)
+_REG_AUDIO_INTERFACE_2 = const(0x09)
+_REG_LEFT_DAC_VOLUME = const(0x0A)
+_REG_RIGHT_DAC_VOLUME = const(0x0B)
+_REG_RESET = const(0x0F)
+_REG_3D_CONTROL = const(0x10)
+_REG_ALC1 = const(0x11)
+_REG_ALC2 = const(0x12)
+_REG_ALC3 = const(0x13)
+_REG_NOISE_GATE = const(0x14)
+_REG_LEFT_ADC_VOLUME = const(0x15)
+_REG_RIGHT_ADC_VOLUME = const(0x16)
+_REG_ADDITIONAL_CONTROL_1 = const(0x17)
+_REG_ADDITIONAL_CONTROL_2 = const(0x18)
+_REG_PWR_MGMT_1 = const(0x19)
+_REG_PWR_MGMT_2 = const(0x1A)
+_REG_ADDITIONAL_CONTROL_3 = const(0x1B)
+_REG_ANTI_POP_1 = const(0x1C)
+_REG_ANTI_POP_2 = const(0x1D)
+_REG_ADCL_SIGNAL_PATH = const(0x20)
+_REG_ADCR_SIGNAL_PATH = const(0x21)
+_REG_LEFT_OUT_MIX = const(0x22)
+_REG_RIGHT_OUT_MIX = const(0x25)
+_REG_MONO_OUT_MIX_1 = const(0x26)
+_REG_MONO_OUT_MIX_2 = const(0x27)
+_REG_LOUT2_VOLUME = const(0x28)
+_REG_ROUT2_VOLUME = const(0x29)
+_REG_MONO_OUT_VOLUME = const(0x2A)
+_REG_INPUT_BOOST_MIXER_1 = const(0x2B)
+_REG_INPUT_BOOST_MIXER_2 = const(0x2C)
+_REG_BYPASS_1 = const(0x2D)
+_REG_BYPASS_2 = const(0x2E)
+_REG_PWR_MGMT_3 = const(0x2F)
+_REG_ADDITIONAL_CONTROL_4 = const(0x30)
+_REG_CLASS_D_CONTROL_1 = const(0x31)
+_REG_CLASS_D_CONTROL_3 = const(0x33)
+_REG_PLL_N = const(0x34)
+_REG_PLL_K_1 = const(0x35)
+_REG_PLL_K_2 = const(0x36)
+_REG_PLL_K_3 = const(0x37)
+
+# Microphone input selections
+MIC_INPUT2 = 0
+MIC_INPUT3 = 1
+MIC_VMID = 2
+
+# Microphone Boost gain options
+MIC_BOOST_GAIN_0DB = 0
+MIC_BOOST_GAIN_13DB = 1
+MIC_BOOST_GAIN_20DB = 2
+MIC_BOOST_GAIN_29DB = 3
+
+'''
+Boost Mixer gain options
+These are used to control the gain (aka volume) at the following settings:
+LIN2BOOST
+LIN3BOOST
+RIN2BOOST
+RIN3BOOST
+'''
+BOOST_MIXER_GAIN_MUTE = 0
+BOOST_MIXER_GAIN_NEG_12DB = 1
+BOOST_MIXER_GAIN_NEG_9DB = 2
+BOOST_MIXER_GAIN_NEG_6DB = 3
+BOOST_MIXER_GAIN_NEG_3DB = 4
+BOOST_MIXER_GAIN_0DB = 5
+BOOST_MIXER_GAIN_3DB = 6
+BOOST_MIXER_GAIN_6DB = 7
+
+# Mic Bias voltage options
+MIC_BIAS_VOLTAGE_0_9_AVDD = 0
+MIC_BIAS_VOLTAGE_0_65_AVDD = 1
+
+# SYSCLK divide
+_SYSCLK_DIV_BY_1 = const(0)
+_SYSCLK_DIV_BY_2 = const(2)
+
+# Gain/Level mins, maxes, offsets and step-sizes
+MIC_GAIN_MIN = -17.25
+MIC_GAIN_MAX = 30.00
+
+ADC_VOLUME_MIN = -97.00
+ADC_VOLUME_MAX = 30.00
+
+DAC_VOLUME_MIN = -127.00
+DAC_VOLUME_MAX = 0.00
+
+ALC_TARGET_MIN = -22.50
+ALC_TARGET_MAX = -1.50
+
+ALC_MAX_GAIN_MIN = -12.00
+ALC_MAX_GAIN_MAX = 30.00
+
+ALC_MIN_GAIN_MIN = -17.25
+ALC_MIN_GAIN_MAX = 24.75
+
+GATE_THRESHOLD_MIN = -76.50
+GATE_THRESHOLD_MAX = -30.00
+
+OUTPUT_VOLUME_MIN = -21.00
+OUTPUT_VOLUME_MAX = 0.00
+
+AMP_VOLUME_MIN = -73.00
+AMP_VOLUME_MAX = 6.00
```

Comment:
I recently switched these back to public to make them accessible if sweeping through the full range of property values. However, these values will be included in the documentation. If you insist, I can revert them back to private.

Also, the `micropython.const` optimization doesn't apply here since it only works on integer values.